stuffer/s2n_stuffer_text.c

```diff
@@ -31,16 +31,14 @@ int s2n_stuffer_peek_char(struct s2n_stuffer *s2n_stuffer, char *c)
 }
 
 /* Peeks in stuffer to see if expected string is present. */
-int s2n_stuffer_peek_check_for_str(struct s2n_stuffer *s2n_stuffer, const char *expected)
+bool s2n_stuffer_peek_check_for_str(struct s2n_stuffer *s2n_stuffer, const char *expected)
 {
-    int orig_read_pos = s2n_stuffer->read_cursor;
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(s2n_stuffer));
+    uint32_t orig_read_pos = s2n_stuffer->read_cursor;
     int rc = s2n_stuffer_read_expected_str(s2n_stuffer, expected);
     s2n_stuffer->read_cursor = orig_read_pos;
-
-    if (rc == 0) {
-        return 1;
-    }
-    return 0;
+    POSTCONDITION_POSIX(s2n_stuffer_is_valid(s2n_stuffer));
```

Comment:
I'm surprised this is legal now.  What does it mean to return -1/0 in boolean context?

Reply:
UB: https://developer.apple.com/documentation/xcode/diagnosing_memory_thread_and_crash_issues_early/invalid_boolean.

If the function is fallible then in must return a error signal (either int or s2n_result).

Reply:
BTW, this is exactly the kind of thing s2n_result is designed to prevent 🙂

Reply:
I changed the function to return an integer value, so we can keep the checks, and I updated the places where the function has been used to properly handle the return value.